src/qt/guiutil.cpp

```diff
@@ -737,33 +737,15 @@ QString formatDurationStr(int secs)
     return strList.join(" ");
 }
 
-QString serviceFlagToStr(const quint64 mask, const int bit)
-{
-    switch (ServiceFlags(mask)) {
-    case NODE_NONE: abort();  // impossible
-    case NODE_NETWORK:         return "NETWORK";
-    case NODE_GETUTXO:         return "GETUTXO";
-    case NODE_BLOOM:           return "BLOOM";
-    case NODE_WITNESS:         return "WITNESS";
-    case NODE_NETWORK_LIMITED: return "NETWORK_LIMITED";
-    // Not using default, so we get warned when a case is missing
-    }
-    if (bit < 8) {
-        return QString("%1[%2]").arg("UNKNOWN").arg(mask);
-    } else {
-        return QString("%1[2^%2]").arg("UNKNOWN").arg(bit);
-    }
-}
-
 QString formatServicesStr(quint64 mask)
 {
     QStringList strList;
 
     for (int i = 0; i < 64; i++) {
-        uint64_t check = 1LL << i;
+        uint64_t check = 1ull << i;
         if (mask & check)
         {
-            strList.append(serviceFlagToStr(check, i));
+            strList.append(QString::fromStdString(serviceFlagToStr(mask, i)));
```

Comment:
Here we should pass `check` instead of `mask`? I did not test it but I think with the current patch `formatServicesStr(NODE_NETWORK | NODE_WITNESS)` will return a string like `UNKNOWN[9] & UNKNOWN[9]`.

The arguments to `serviceFlagToStr()` are redundant (one can be derived from the other easily) which I think is confusing and could lead to slippages. I agree with @laanwj that just one of them should be passed.